File: PM_Case_Managemnt_API/Models/PM/ActivityProgress.cs

using PM_Case_Managemnt_API.Models.Common;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PM_Case_Managemnt_API.Models.PM
{
    public class ActivityProgress : CommonModel
    {
        public ActivityProgress()
        {
            ProgressAttachments = new HashSet<ProgressAttachment>();
        }

        public Guid ActivityId { get; set; }
        public virtual Activity Activity { get; set; }
        public float ActualBudget { get; set; }
        public float ActualWorked { get; set; }
        public Guid EmployeeValueId { get; set; }
        public virtual Employee EmployeeValue { get; set; } = null!;
        public Guid QuarterId { get; set; }
        public virtual ActivityTargetDivision quarter { get; set; } = null!;
        public string DocumentPath { get; set; } = null!;
        public string FinanceDocumentPath { get; set; } = null!;
        public approvalStatus IsApprovedByCoordinator { get; set; }
        public approvalStatus IsApprovedByFinance { get; set; }
        public approvalStatus IsApprovedByDirector { get; set; }
        public string FinanceApprovalRemark { get; set; } = null!;
        public string CoordinatorApprovalRemark { get; set; } = null!;
        public string DirectorApprovalRemark { get; set; } = null!;
        public string Lat { get; set; } = null!;
        public string Lng { get; set; } = null!;
        public ProgressStatus progressStatus { get; set; }

     
        public  ICollection<ProgressAttachment> ProgressAttachments { get; set; }
    }
    public enum ProgressStatus
    {
        SimpleProgress,
        Finalize
    }

	public enum approvalStatus
	{
		pending ,
		approved ,
		rejected
	}
}

File: PM_Mobile/daf_project1/lib/models/activity_progress.dart
import 'employee.dart';

class ActivityProgress{
  late String _id;
  late double _actualBudget;
  late double _actualWorked;
  late String? _documentPath;
  late bool _isApprovedByCoordinator;
  late bool _isApprovedByFinance;
  late String _employeeId;
  late Employee _employee;
  late String _remark;

  String get id => _id;
  ActivityProgress();

  ActivityProgress.fromJson(Map<String,dynamic> json):
    _id = json['id'],
    _actualBudget = double.parse((json['actualBudget']??0).toString()),
    _actualWorked = double.parse((json['actualWorked']??0).toString()),
    _employeeId = json['employeeValueId'],
    _employee = Employee.fromJson(json['employeeValue']),
    _documentPath= json['documentPath'],
    _isApprovedByCoordinator = json['isApprovedByCoordinator'],
    _isApprovedByFinance = json['isApprovedByFinance'];


  @override
  String toString() {
    return _id;
  }

  double get actualBudget => _actualBudget;

  String get remark => _remark;

  Employee get employee => _employee;

  String get employeeId => _employeeId;

  bool get isApprovedByFinance => _isApprovedByFinance;

  bool get isApprovedByCoordinator => _isApprovedByCoordinator;

  String? get documentPath => _documentPath;

  double get actualWorked => _actualWorked;
}